### ISSUE TITLE
setFilterModel along with applyColumnState causes erratic behavior

### ISSUE THREAD

**Issue by ehsanshahidi** [ag-grid/ag-grid#5152]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

Please see this [Plunkr page.](https://plnkr.co/edit/Ax2xL0xHH4VqHXFF?open=index.tsx) and take the following steps to reproduce:

- Open the [plunkr](https://plnkr.co/edit/Ax2xL0xHH4VqHXFF?open=index.tsx)
- Group by Country, then by Sport
- Filter the Year column to only show 2002
- Optional: Make the Group column (first column) wider
- Save the state using the provided button (this saves the column state and the filter model)
- Reset the state using the provided button (this resets the column state and the filter model)
- Restore the state using the provided button (this restores the saved column state and filter model)
- Note the seemingly missing rows.
- Try to expand the "Netherlands (6)" row
- Note the overlapping text

**Current behavior**
When the above steps are taken, after restoring the column state and filter model, some rows appear empty, and when the row below an empty one is expanded, texts seem to overlap. This doesn't happen if only column state is saved and restored. This also doesn't seem to happen when you just group and filter without saving and loading the respective states.

**Expected behavior**
The expected behavior is that the grouped rows reflect the loaded filter after the state of columns and filter is restored. 

**Please tell us about your environment:**
NA - plunkr

* **AG Grid version:** X.X.X
NA - plunkr

* **Browser:**
Chrome/101.0.4951.54

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript - plunkr

Thank you for the great product!

**Comment by makinggoodsoftware** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by ehsanshahidi** [comment]
@makinggoodsoftware Thanks for the update. 
For others who may be blocked by this, a workaround is doing a `gridOptions.api.redrawRows()` after loading the column state and filter model.

**Comment by kiril-matev** [comment]
Hello @ehsanshahidi !

Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

We have added this bug to our backlog and we are tracking it with the following reference and description:
**AG-6805 When using React, applying column state and then filter model produces empty lines inside the grid rows**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

You can see the next release date on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

**Now that this issue is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this bug is fixed in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/**

Thanks again for bringing this up with us - we appreciate your time!

**Comment by ehsanshahidi** [comment]
Thanks for the update @kiril-matev. Much appreciated.

**Comment by makinggoodsoftware** [comment]
Hi,

In order to help us keep our issues queue clear, we are going to close this issue

The official way to check the status for an issue raised here is to check its AG-xxxx ID against our pipeline and changelog

…the pipeline (if not yet developed)

https://www.ag-grid.com/ag-grid-pipeline/

... or the changelog, (if the issue has been resolved)

https://www.ag-grid.com/ag-grid-changelog/

We keep issues raised in the pipeline through github open for a while so that the community can add some feedback to them, and close them when there is no activity for a while.

If you want to add further feedback to a closed ticket with this comment, we would recommend opening a new ticket as we don’t monitor closed issues.

**Comment by bennyperl** [comment]
It's seems that filter works (the row gets index "0" for sorting  ) but still has Y offset by "translateY(825px)"
